pandas/core/indexes/period.py

```diff
@@ -613,25 +611,21 @@ def get_loc(self, key, method=None, tolerance=None):
 
         try:
             key = Period(key, freq=self.freq)
+            ordinal = key.ordinal if key is not NaT else key.value
         except ValueError:
             # we cannot construct the Period
             # as we have an invalid type
             if is_list_like(key):
                 raise TypeError(f"'{key}' is an invalid key")
             raise KeyError(key)
 
-        ordinal = key.ordinal if key is not NaT else key.value
+        if tolerance is not None:
+            tolerance = self._convert_tolerance(tolerance, np.asarray(key))
+
         try:
-            return self._engine.get_loc(ordinal)
+            return self._int64index.get_loc(ordinal, method, tolerance)
         except KeyError:
-
-            try:
-                if tolerance is not None:
-                    tolerance = self._convert_tolerance(tolerance, np.asarray(key))
-                return self._int64index.get_loc(ordinal, method, tolerance)
-
-            except KeyError:
-                raise KeyError(key)
+            raise KeyError(key)
```

Comment:
going directly to _int64index does simplify the code here which is nice, but i imagine involves a perf penalty for by-far-the-most-common case with method=tolerance=None.  Am I wrong about that?

Reply:
@jbrockmendel 
Yes, you are right. I pointed out the wrong key. Like #31021, cleaning up codes is best choice.
Is it okay if  i close PR and issue?

Reply:
sure